### PR TITLE
Tag problem panel

### PR DESCRIPTION
**Pull request by geniusgordon**
add tag filter in problem panel

### CODE REVIEW

File: problem/forms.py [hunk 1]
```diff
@@ -134,3 +134,17 @@ class Meta:
...
 
+
+class TagFilter(forms.Form):
+    tag_name = forms.CharField(max_length=20, label='Tag', required=False,
+            widget= autocomplete_light.TextWidget('TagAutocomplete',
+                                        attrs={'class': 'form-control'}))
+
+    def clean_tag_name(self):
+        tag_name = self.cleaned_data['tag_name'].strip()
+        if len(tag_name) == 0:
```
Comment (nickelcheng) [inline problem/forms.py]: Do you mean return an empty string?
I don't understand why return when the length is equal to 0

Comment (geniusgordon) [inline problem/forms.py]: fixed

File: problem/views.py [hunk 2]
```diff
@@ -49,25 +49,32 @@ def problem(request):
...
-            p.not_pass_rate = 100.0 - p.pass_rate
-            p.pass_rate = "%.2f" % (p.pass_rate)
-            p.not_pass_rate = "%.2f" % (p.not_pass_rate)
-        else:
-            p.no_submission = True
-
     return render_index(request, 'problem/panel.html',
                   {'all_problem': problems, 'mine': mine,
-                   'can_add_problem': can_add_problem})
+                      'can_add_problem': can_add_problem, 'tag_filter': tag_filter})
```
Comment (nickelcheng) [inline problem/views.py]: I prefer ```'can_add_problem'``` align with ```'all_problem'```

**Comment by nickelcheng** [comment]
Please add link on tag name (in problem page)

**Comment by nickelcheng** [comment]
Your link leads to origin problem panel (all problems are listed).

File: problem/views.py (hunk 2 above)
Comment (geniusgordon) [inline problem/views.py]: fixed

**Comment by CHIAHSINHOU** [comment]
I have a question. If we have a contest, and the problem A with tag DFS is in the contest, should the problem be select when we use tag search to search DFS?.

**Comment by nickelcheng** [comment]
I think the problem should not show on search list.
@geniusgordon please exclude them.

**Comment by geniusgordon** [comment]
done

**Comment by nickelcheng** [comment]
+1

**Comment by CHIAHSINHOU** [comment]
Would you mind change the search layout?
Maybe move the txt field to the left hand side.

**Comment by geniusgordon** [comment]
done

**Comment by CHIAHSINHOU** [comment]
This PR is ok for me.

**Comment by nickelcheng** [comment]
Going to merge

**Pull request merged**